python/graphstorm/model/gnn_encoder_base.py

```diff
@@ -102,6 +102,23 @@ def dist_inference(self, g, get_input_embeds, batch_size, fanout,
         return dist_inference(g, self, get_input_embeds, batch_size, fanout,
                             edge_mask=edge_mask, task_tracker=task_tracker)
 
+def prepare_for_wholegraph(g, input_nodes):
+    """ Add missing ntypes in input_nodes for wholegraph compatibility
+
+    Parameters
+    ----------
+    g : DistGraph
+        Input graph
+    input_nodes : Tensor
+        Input nodes retrieved from the dataloder
+    """
+    if input_nodes:
+        tmp_keys = [ntype for ntype in g.ntypes if ntype not in input_nodes]
+        input_nodes.update({ntype: th.empty((0,), dtype=g.idtype) \
+            for ntype in tmp_keys})
+    else:
```

Comment:
According to https://github.com/awslabs/graphstorm/pull/598/files#diff-4f7c660635ed15aa4772344912243762fd44a6af1a4b8fa20152d9fb63a1c49aR212, input_nodes will never be None.

Reply:
Further more, with `input_nodes = {ntype: th.empty((0,), dtype=g.idtype) for ntype in g.ntypes}`, how do you return the newly created input_nodes?

Reply:
Fixed.